### PR TITLE
more strict typing patterns for route navigation

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
The previous navigation pattern was very bug-prone. It accepted any string.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -96,14 +96,21 @@
...
-    "eslint-config-airbnb": "^18.2.0",
+    "eslint-config-airbnb": "^18.2.1",
     "gh-pages": "^3.1.0",
     "husky": "^4.2.5",
     "lint-staged": "^10.2.11",
     "postcss": "^7.0.35",
     "prettier": "^2.0.5",
     "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.1.2",
-    "typescript": "^3.9.7"
+    "typescript": "^4.4.4"
```
Comment (gabrielsaruhashi) [inline package.json:109]: template literals were introduced on TS 4.1

File: package.json [hunk 2]
```diff
@@ -96,14 +96,21 @@
...
+  },
+  "resolutions": {
+    "**/typescript": "^4.4.4",
+    "**/@typescript-eslint/eslint-plugin": "^4.15.1",
+    "**/@typescript-eslint/parser": "^4.15.1"
```
Comment (cainlevy) [inline package.json:114]: 1) wait this works with npm??
2) why not just specify these versions in the dependencies?

File: src/constants/Routes.ts [hunk 3]
```diff
@@ -50,5 +76,10 @@ export const ROUTES: Route[] = [
   { path: "/admin", component: AdminPage, label: "Admin" },
   { path: "/call/:id", component: RecordingPage, label: "Past Call" },
   { path: "/visitors", component: VisitorsPage, label: "Visitors" },
+  {
+    path: "/message-review",
+    component: MessageReviewPage,
+    label: "Review Messages",
+  },
```
Comment (cainlevy) [inline src/constants/Routes.ts:83]: feature flagged?

**Review by cainlevy — APPROVED** [review]
cool use of template literal types

File: package.json (hunk 2 above)
Comment (gabrielsaruhashi) [reply package.json:114]: I was struggling to get template literals to compile 😱 I thought this had resolved it but updating typescript to the right version was the right answer. 
Thanks for calling this out

**Pull request merged**